src/marqo/core/monitoring/memory_profiler.py

```diff
@@ -0,0 +1,25 @@
+import gc
+import tracemalloc
+
+from memory_profiler import memory_usage
+
+from marqo.core.models.memory_profile import MemoryProfile
+
+
+def get_memory_profile() -> MemoryProfile:
+    tracemalloc.start()
+
+    snapshot = tracemalloc.take_snapshot()
+    stats = snapshot.statistics('lineno')
+
+    # GC
+    gc.collect()  # Force garbage collection
```

Comment:
does it makes sense for the the get_profile function to also do a GC? doesn't that mean the profiled memory is accurate? 
Also, I am wary of a read call (like get_mem_profile) with a side effect like this. 
Perhaps a dedicated gc() function is better